sdk/database/helper/connutil/cloudsql.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package connutil
+
+import (
+	"fmt"
+
+	"cloud.google.com/go/cloudsqlconn"
+	"cloud.google.com/go/cloudsqlconn/postgres/pgxv4"
+	"cloud.google.com/go/cloudsqlconn/sqlserver/mssql"
+)
+
+func (c *SQLConnectionProducer) getCloudSQLDriverName() (string, error) {
+	var driverName string
+	switch c.Type {
+	case dbTypeMSSQL:
+		driverName = cloudSQLMSSQL
+	case dbTypePostgres:
+		driverName = cloudSQLPostgres
+	default:
+		return "", fmt.Errorf("unrecognized DB type: %s", c.Type)
+	}
+
+	return driverName, nil
+}
+
+func (c *SQLConnectionProducer) registerDrivers(driverName string, credentials string) (func() error, error) {
+	typ, err := c.getCloudSQLDriverName()
+	if err != nil {
+		return nil, err
+	}
+
+	opts, err := GetCloudSQLAuthOptions(credentials)
+	if err != nil {
+		return nil, err
+	}
+
+	switch typ {
+	case cloudSQLMSSQL:
+		return mssql.RegisterDriver(driverName, opts...)
+	case cloudSQLPostgres:
+		return pgxv4.RegisterDriver(driverName, opts...)
+	}
+
+	return nil, fmt.Errorf("unrecognized cloudsql type encountered: %s", typ)
+}
+
+// GetCloudSQLAuthOptions takes a credentials (file) or a credentialsJSON (the actual data) and returns
```

Comment:
`credentials` can also be a path to the credentials file, or does the driver ingest it from a known location? The way that the options are being provided (and the fact that `WithIAMAuthN` does not have any input params) seems like it's the latter.

Reply:
@vinay-gopalan would know better, but I was under the impression that we'd always get bytes (i.e., a json) at this point, so this comment should probably be updated.

My understanding is that users have the option of using '@' notation in the cli (and in curl), which will read the file and handle sending the bytes.

Reply:
I updated the comment on the assumption that it is always the JSON now, since that's how we're treating it.